paramiko/pkey.py

```diff
@@ -43,6 +44,27 @@
 from paramiko.message import Message
 
 
+# Temporary patch handling the correct package import path for decrepit
+# algorithms. Ideally would bump the minimum required cryptography version
+# in `setup.py` for a new major paramiko release. This is only needed as
+# long as paramiko still has support for TripleDES
+try:
+    from cryptography.hazmat.primitives.ciphers.algorithms import TripleDES
+except ImportError:
+    from cryptography.hazmat.decrepit.ciphers.algorithms import TripleDES
+
+    warnings.warn(
+        """
+        Paramiko depends on cryptography and leverages some of the ciphers
+        defined by it. However, since the release of cryptography>=43.0.0,
+        TripleDES cipher is soft-deprecated and support for it in will be
+        dropped in 48.0.0. Please consider alternative ciphers
+
+        Source: https://github.com/pyca/cryptography/commit/722a6393e61b3ac
+        """
+    )
```

Comment:
It'd be better to invert the order here: import it from the _new_ name, and if that's not available then fall back to the old name.

There's no need to emit a warning here -- TripleDES's location is changing, it's not being removed.